boost-maven/boost-maven-plugin/src/main/java/io/openliberty/boost/maven/utils/MavenProjectUtil.java

```diff
@@ -51,4 +55,42 @@ public static String findSpringBootVersion(MavenProject project) {
         return dependencies;
     }
 
+    public static String getJavaCompilerTargetVersion(MavenProject project) {
+
+        Properties mavenProperties = project.getProperties();
+
+        for (Map.Entry<Object, Object> property : mavenProperties.entrySet()) {
+            String propertyKey = property.getKey().toString();
+            String propertyValue = property.getValue().toString();
+
+            if (propertyKey.equals("maven.compiler.target") || propertyKey.equals("maven.compiler.release")) {
+                return propertyValue;
+            }
+        }
+
+        List<Plugin> plugins = project.getBuildPlugins();
+
+        for (Plugin plugin : plugins) {
+            if (plugin.getArtifactId().equals("maven-compiler-plugin")) {
+                Xpp3Dom config = (Xpp3Dom) plugin.getConfiguration();
+
+                if (config != null) {
+                    Xpp3Dom release = config.getChild("release");
+
+                    if (release != null) {
+                        return release.getValue();
+                    }
+
+                    Xpp3Dom target = config.getChild("target");
+
+                    if (target != null) {
+                        return target.getValue();
+                    }
+                }
+            }
+        }
+
+        return "1.6";
```

Comment:
How do we know that this is the default value?

Reply:
Ah good catch... the default is 1.6 for maven-compiler-plugin version  3.8.0. (latest version).  previous versions should be 1.5.  I'll have to check to see what maven version they are using.

Reply:
Thinking about this more, I don't think it's worth going below 1.6. I believe Liberty requires a minimum jre of 1.6, so other things might break if the app is run on 1.5 or lower.

Reply:
Actually, I think the minimum  is Java 7 now. For Open Liberty it may even be Java 8.

Reply:
I changed this method to just return the compiler target if it is explicitly configured. Otherwise it will return null.  The JDBC booster will then determine what feature to use. That's also changed to use jdbc-4.1 (SE 7 spec) as a default or for any compiler target value less than 1.7. It's unlikely that someone would be using a target of less than this, and even still, since Liberty wont run (or at least isnt supported) at less than 1.7, there should be little risk to having jdbc-4.1 as a minimum.